--- src/Perlang.Common/PerlangCli.cs	2026-09-12 22:42:10.868545135 +0300
+++ src/Perlang.Common/PerlangCli.cs-edited	2026-09-12 22:42:10.864545226 +0300
@@ -587,27 +587,27 @@
         internal int current;
         internal int line;
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner5matchEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner5matchEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool Match(__IntPtr __instance, char expected);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsAlpha(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsUnderscore(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner16is_alpha_numericEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner16is_alpha_numericEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsAlphaNumeric(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_digitEDsN16NumericTokenBase16NumericTokenBaseE", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_digitEDsN16NumericTokenBase16NumericTokenBaseE", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsDigit(char c, global::Perlang.NumericTokenBase @base);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner7advanceEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner7advanceEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char Advance(__IntPtr __instance);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner12advance_lineEv", CallingConvention = __CallingConvention.Cdecl)]
@@ -616,10 +616,10 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner20set_start_to_currentEv", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern void SetStartToCurrent(__IntPtr __instance);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner4peekEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner4peekEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char Peek(__IntPtr __instance);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9peek_nextEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9peek_nextEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char PeekNext(__IntPtr __instance);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9is_at_endEv", CallingConvention = __CallingConvention.Cdecl)]
@@ -648,7 +648,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::PerlangScanner managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -863,7 +863,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -950,7 +950,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Token managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -1113,7 +1113,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -1236,7 +1236,7 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "native_main", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern void NativeMain(int argc, sbyte** argv);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z20create_numeric_tokenPKcS0_iS0_DsbbN16NumericTokenBase16NumericTokenBaseEi", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z20create_numeric_tokenPKcS0_iS0_DsbbN16NumericTokenBase16NumericTokenBaseEi", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern __IntPtr CreateNumericToken([MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string number_characters, char suffix, bool has_suffix, bool is_fractional, global::Perlang.NumericTokenBase number_base, int number_styles);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z16as_numeric_tokenP6IToken", CallingConvention = __CallingConvention.Cdecl)]
@@ -1245,7 +1245,7 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z32get_numeric_token_literal_stringP6IToken", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern __IntPtr GetNumericTokenLiteralString(__IntPtr token);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z24get_numeric_token_suffixP6IToken", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z24get_numeric_token_suffixP6IToken", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char GetNumericTokenSuffix(__IntPtr token);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z22create_perlang_scannerPKc", CallingConvention = __CallingConvention.Cdecl)]
@@ -1257,7 +1257,7 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z19create_string_tokenN9TokenType9TokenTypeEPKcS2_S2_i", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern __IntPtr CreateStringToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern __IntPtr CreateCharToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, char literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_null_tokenN9TokenType9TokenTypeEPKcS2_i", CallingConvention = __CallingConvention.Cdecl)]
@@ -1306,7 +1306,7 @@
         __Internal.NativeMain(argc, argv);
     }
 
-    public static global::IIToken CreateNumericToken(string lexeme, string file_name, int line, string number_characters, char suffix, bool has_suffix, bool is_fractional, global::Perlang.NumericTokenBase number_base, int number_styles)
+    public static global::IToken CreateNumericToken(string lexeme, string file_name, int line, string number_characters, char suffix, bool has_suffix, bool is_fractional, global::Perlang.NumericTokenBase number_base, int number_styles)
     {
         var ___ret = __Internal.CreateNumericToken(lexeme, file_name, line, number_characters, suffix, has_suffix, is_fractional, number_base, number_styles);
         var __result0 = global::IToken.__GetOrCreateInstance(___ret, true);
@@ -1348,21 +1348,21 @@
         __Internal.DeletePerlangScanner(__arg0);
     }
 
-    public static global::IIToken CreateStringToken(global::Perlang.TokenType token_type, string lexeme, string literal, string file_name, int line)
+    public static global::IToken CreateStringToken(global::Perlang.TokenType token_type, string lexeme, string literal, string file_name, int line)
     {
         var ___ret = __Internal.CreateStringToken(token_type, lexeme, literal, file_name, line);
         var __result0 = global::IToken.__GetOrCreateInstance(___ret, true);
         return __result0;
     }
 
-    public static global::IIToken CreateCharToken(global::Perlang.TokenType token_type, string lexeme, char literal, string file_name, int line)
+    public static global::IToken CreateCharToken(global::Perlang.TokenType token_type, string lexeme, char literal, string file_name, int line)
     {
         var ___ret = __Internal.CreateCharToken(token_type, lexeme, literal, file_name, line);
         var __result0 = global::IToken.__GetOrCreateInstance(___ret, true);
         return __result0;
     }
 
-    public static global::IIToken CreateNullToken(global::Perlang.TokenType token_type, string lexeme, string file_name, int line)
+    public static global::IToken CreateNullToken(global::Perlang.TokenType token_type, string lexeme, string file_name, int line)
     {
         var ___ret = __Internal.CreateNullToken(token_type, lexeme, file_name, line);
         var __result0 = global::IToken.__GetOrCreateInstance(___ret, true);
@@ -1565,7 +1565,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IObject managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -1703,7 +1703,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -2725,7 +2725,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IntArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -3443,7 +3443,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.ObjectArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -3559,12 +3559,6 @@
             [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4CharC2ERKS0_", CallingConvention = __CallingConvention.Cdecl)]
             internal static extern void cctor(__IntPtr __instance, __IntPtr _0);
 
-            [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4Char8to_upperEDs", CallingConvention = __CallingConvention.Cdecl)]
-            internal static extern char ToUpper(char literal);
-
-            [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4Char8to_lowerEDs", CallingConvention = __CallingConvention.Cdecl)]
-            internal static extern char ToLower(char literal);
-
             [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZNK7perlang4Char5valueEv", CallingConvention = __CallingConvention.Cdecl)]
             internal static extern char Value(__IntPtr __instance);
         }
@@ -3658,18 +3652,6 @@
             __Instance = IntPtr.Zero;
         }
 
-        public static char ToUpper(char literal)
-        {
-            var ___ret = __Internal.ToUpper(literal);
-            return ___ret;
-        }
-
-        public static char ToLower(char literal)
-        {
-            var ___ret = __Internal.ToLower(literal);
-            return ___ret;
-        }
-
         public char Value
         {
             get
@@ -3699,7 +3681,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -3790,7 +3772,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.PerlangType managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -3924,7 +3906,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.StringArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -4623,7 +4605,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringHasher managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -4729,7 +4711,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringComparer managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -4854,7 +4836,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.MutableStringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -4986,7 +4968,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringArray managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -5074,7 +5056,7 @@
             {
                 get
                 {
-                    return (string*) ((__Internal*)__Instance)->items;
+                    return (sbyte**) ((__Internal*)__Instance)->items;
                 }
             }
 
@@ -5134,7 +5116,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -5305,7 +5287,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IllegalStateException managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -5412,7 +5394,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -5498,7 +5480,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Io.File managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -5849,7 +5831,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Text.StringBuilder managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -6068,7 +6050,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Stdlib.Base64 managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -6191,7 +6173,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -6345,7 +6327,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::MutableStringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
