--- src/Perlang.Common/PerlangCli.cs	2025-10-31 21:51:57.317507472 +0200
+++ src/Perlang.Common/PerlangCli.cs-edited	2025-10-31 21:51:57.313507440 +0200
@@ -23,11 +23,11 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScannerC2Ev", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern void ctor(__IntPtr __instance);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsAlpha(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsUnderscore(char c);
     }
@@ -44,7 +44,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::PerlangScanner managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -181,7 +181,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -273,7 +273,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Token managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -406,7 +406,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -482,7 +482,7 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z19create_string_tokenN9TokenType9TokenTypeEPKcS2_S2_i", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern __IntPtr CreateStringToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern __IntPtr CreateCharToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, char literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_null_tokenN9TokenType9TokenTypeEPKcS2_i", CallingConvention = __CallingConvention.Cdecl)]
@@ -728,7 +728,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IObject managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -866,7 +866,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -989,7 +989,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IntArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -1133,7 +1133,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.ObjectArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -1371,7 +1371,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -1462,7 +1462,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.PerlangType managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -1596,7 +1596,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.StringArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -1711,7 +1711,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringHasher managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -1817,7 +1817,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringComparer managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -1942,7 +1942,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.MutableStringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -2074,7 +2074,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringArray managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -2162,7 +2162,7 @@
             {
                 get
                 {
-                    return (string*) ((__Internal*)__Instance)->items;
+                    return (sbyte**) ((__Internal*)__Instance)->items;
                 }
             }
 
@@ -2222,7 +2222,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -2385,7 +2385,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IllegalStateException managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -2492,7 +2492,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -2578,7 +2578,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Io.File managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -2705,7 +2705,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Text.StringBuilder managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -2916,7 +2916,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Stdlib.Base64 managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -3039,7 +3039,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -3193,7 +3193,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::MutableStringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
