--- src/Perlang.Common/PerlangCli.cs	2026-01-20 22:56:38.397451693 +0200
+++ src/Perlang.Common/PerlangCli.cs-edited	2026-01-20 22:56:38.393451779 +0200
@@ -24,19 +24,19 @@
         internal int current;
         internal int line;
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner5matchEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner5matchEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool Match(__IntPtr __instance, char expected);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsAlpha(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsUnderscore(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner7advanceEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner7advanceEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char Advance(__IntPtr __instance);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner12advance_lineEv", CallingConvention = __CallingConvention.Cdecl)]
@@ -45,10 +45,10 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner20set_start_to_currentEv", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern void SetStartToCurrent(__IntPtr __instance);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner4peekEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner4peekEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char Peek(__IntPtr __instance);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9peek_nextEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9peek_nextEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char PeekNext(__IntPtr __instance);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9is_at_endEv", CallingConvention = __CallingConvention.Cdecl)]
@@ -77,7 +77,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::PerlangScanner managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -280,7 +280,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -372,7 +372,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Token managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -505,7 +505,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -587,7 +587,7 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z19create_string_tokenN9TokenType9TokenTypeEPKcS2_S2_i", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern __IntPtr CreateStringToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern __IntPtr CreateCharToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, char literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_null_tokenN9TokenType9TokenTypeEPKcS2_i", CallingConvention = __CallingConvention.Cdecl)]
@@ -847,7 +847,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IObject managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -985,7 +985,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -1868,7 +1868,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IntArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -2175,7 +2175,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.ObjectArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -2291,12 +2291,6 @@
             [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4CharC2ERKS0_", CallingConvention = __CallingConvention.Cdecl)]
             internal static extern void cctor(__IntPtr __instance, __IntPtr _0);
 
-            [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4Char8to_upperEDs", CallingConvention = __CallingConvention.Cdecl)]
-            internal static extern char ToUpper(char literal);
-
-            [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4Char8to_lowerEDs", CallingConvention = __CallingConvention.Cdecl)]
-            internal static extern char ToLower(char literal);
-
             [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZNK7perlang4Char5valueEv", CallingConvention = __CallingConvention.Cdecl)]
             internal static extern char Value(__IntPtr __instance);
         }
@@ -2390,18 +2384,6 @@
             __Instance = IntPtr.Zero;
         }
 
-        public static char ToUpper(char literal)
-        {
-            var ___ret = __Internal.ToUpper(literal);
-            return ___ret;
-        }
-
-        public static char ToLower(char literal)
-        {
-            var ___ret = __Internal.ToLower(literal);
-            return ___ret;
-        }
-
         public char Value
         {
             get
@@ -2431,7 +2413,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -2522,7 +2504,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.PerlangType managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -2656,7 +2638,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.StringArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -3085,7 +3067,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringHasher managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -3191,7 +3173,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringComparer managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -3316,7 +3298,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.MutableStringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -3448,7 +3430,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringArray managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -3536,7 +3518,7 @@
             {
                 get
                 {
-                    return (string*) ((__Internal*)__Instance)->items;
+                    return (sbyte**) ((__Internal*)__Instance)->items;
                 }
             }
 
@@ -3596,7 +3578,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -3759,7 +3741,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IllegalStateException managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -3866,7 +3848,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -3952,7 +3934,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Io.File managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -4079,7 +4061,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Text.StringBuilder managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -4290,7 +4272,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Stdlib.Base64 managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -4413,7 +4395,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -4567,7 +4549,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::MutableStringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
