--- src/Perlang.Common/PerlangCli.cs	2026-09-08 12:56:58.594344570 +0300
+++ src/Perlang.Common/PerlangCli.cs-edited	2026-09-08 12:56:58.594344570 +0300
@@ -24,27 +24,27 @@
         internal int current;
         internal int line;
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner5matchEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner5matchEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool Match(__IntPtr __instance, char expected);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_alphaEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsAlpha(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner13is_underscoreEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsUnderscore(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner16is_alpha_numericEDs", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner16is_alpha_numericEDs", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsAlphaNumeric(char c);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_digitEDsN16NumericTokenBase16NumericTokenBaseE", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner8is_digitEDsN16NumericTokenBase16NumericTokenBaseE", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         [return: MarshalAs(UnmanagedType.I1)]
         internal static extern bool IsDigit(char c, global::Perlang.NumericTokenBase @base);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner7advanceEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner7advanceEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char Advance(__IntPtr __instance);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner12advance_lineEv", CallingConvention = __CallingConvention.Cdecl)]
@@ -53,10 +53,10 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner20set_start_to_currentEv", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern void SetStartToCurrent(__IntPtr __instance);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner4peekEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner4peekEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char Peek(__IntPtr __instance);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9peek_nextEv", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9peek_nextEv", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern char PeekNext(__IntPtr __instance);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN14PerlangScanner9is_at_endEv", CallingConvention = __CallingConvention.Cdecl)]
@@ -85,7 +85,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::PerlangScanner managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -300,7 +300,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -392,7 +392,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Token managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -525,7 +525,7 @@
         private static CppSharp.Runtime.VTables VTables;
         private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
             SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
         static VTableLoader()
         {
             _dtorDelegateInstance += _dtorDelegateHook;
@@ -607,7 +607,7 @@
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z19create_string_tokenN9TokenType9TokenTypeEPKcS2_S2_i", CallingConvention = __CallingConvention.Cdecl)]
         internal static extern __IntPtr CreateStringToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
-        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl)]
+        [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_char_tokenN9TokenType9TokenTypeEPKcDsS2_i", CallingConvention = __CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         internal static extern __IntPtr CreateCharToken(global::Perlang.TokenType token_type, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string lexeme, char literal, [MarshalAs(UnmanagedType.CustomMarshaler, MarshalTypeRef = typeof(CppSharp.Runtime.UTF8Marshaller))] string file_name, int line);
 
         [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_Z17create_null_tokenN9TokenType9TokenTypeEPKcS2_i", CallingConvention = __CallingConvention.Cdecl)]
@@ -886,7 +886,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IObject managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -1024,7 +1024,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -2046,7 +2046,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IntArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -2764,7 +2764,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.ObjectArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -2880,12 +2880,6 @@
             [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4CharC2ERKS0_", CallingConvention = __CallingConvention.Cdecl)]
             internal static extern void cctor(__IntPtr __instance, __IntPtr _0);
 
-            [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4Char8to_upperEDs", CallingConvention = __CallingConvention.Cdecl)]
-            internal static extern char ToUpper(char literal);
-
-            [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZN7perlang4Char8to_lowerEDs", CallingConvention = __CallingConvention.Cdecl)]
-            internal static extern char ToLower(char literal);
-
             [SuppressUnmanagedCodeSecurity, DllImport("perlang_cli", EntryPoint = "_ZNK7perlang4Char5valueEv", CallingConvention = __CallingConvention.Cdecl)]
             internal static extern char Value(__IntPtr __instance);
         }
@@ -2979,18 +2973,6 @@
             __Instance = IntPtr.Zero;
         }
 
-        public static char ToUpper(char literal)
-        {
-            var ___ret = __Internal.ToUpper(literal);
-            return ___ret;
-        }
-
-        public static char ToLower(char literal)
-        {
-            var ___ret = __Internal.ToLower(literal);
-            return ___ret;
-        }
-
         public char Value
         {
             get
@@ -3020,7 +3002,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -3111,7 +3093,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.PerlangType managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -3245,7 +3227,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.StringArray managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -3944,7 +3926,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringHasher managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -4050,7 +4032,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringComparer managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -4175,7 +4157,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.MutableStringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -4307,7 +4289,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringArray managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -4395,7 +4377,7 @@
             {
                 get
                 {
-                    return (string*) ((__Internal*)__Instance)->items;
+                    return (sbyte**) ((__Internal*)__Instance)->items;
                 }
             }
 
@@ -4455,7 +4437,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Collections.StringHashSet managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -4626,7 +4608,7 @@
 
         internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.IllegalStateException managed)
         {
-    
+
             return NativeToManagedMap.TryGetValue(native, out managed);
         }
 
@@ -4733,7 +4715,7 @@
             private static CppSharp.Runtime.VTables VTables;
             private static readonly global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>
                 SafeHandles = new global::System.Collections.Generic.List<CppSharp.Runtime.SafeUnmanagedMemoryHandle>();
-                
+
             static VTableLoader()
             {
                 _dtorDelegateInstance += _dtorDelegateHook;
@@ -4819,7 +4801,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Io.File managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -5170,7 +5152,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Text.StringBuilder managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -5389,7 +5371,7 @@
 
             internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::Perlang.Stdlib.Base64 managed)
             {
-    
+
                 return NativeToManagedMap.TryGetValue(native, out managed);
             }
 
@@ -5512,7 +5494,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::StringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
@@ -5666,7 +5648,7 @@
 
     internal static bool __TryGetNativeToManagedMapping(IntPtr native, out global::MutableStringTokenTypeDictionary managed)
     {
-    
+
         return NativeToManagedMap.TryGetValue(native, out managed);
     }
 
